lib/Backend/Lower.cpp

```diff
@@ -3993,7 +3993,22 @@ IR::JnHelperMethod GetArrayAllocMemHelper<Js::JavascriptNativeFloatArray>()
 
 template <typename ArrayType>
 IR::RegOpnd *
-Lowerer::GenerateArrayAlloc(IR::Instr *instr, uint32 * psize, Js::ArrayCallSiteInfo * arrayInfo, bool * pIsHeadSegmentZeroed, bool isArrayObjCtor /* = false */)
+Lowerer::GenerateArrayLiteralsAlloc(IR::Instr *instr, uint32 * psize, Js::ArrayCallSiteInfo * arrayInfo, bool * pIsHeadSegmentZeroed)
```

Comment:
This method is used to generate memory allocation for array literals. I kept it this way for easy readability. I don't like having 2 defaults because then caller has to make sure that it is calling the method with right parameters.

---
In reply to: [95441552](https://github.com/Microsoft/ChakraCore/pull/2325#discussion_r95441552) [](ancestors = 95441552)